src/ORM/DataList.php

```diff
@@ -1480,8 +1545,11 @@ public function eagerLoad(...$relationChains): static
                 // Keep track of what we've seen before so we don't accidentally add a level 1 relation
                 // (e.g. "Players") to the chains list when we already have it as part of a longer chain
                 // (e.g. "Players.Teams")
-                $list->eagerLoadAllRelations[$item] = $item;
+                $list->eagerLoadAllRelations[$item] ??= null;
```

Comment:
This change does two things:
1. Default the value to `null`, so we can do a simply nullable check to see if there was a callback set for that chain or not
2. Doesn't override existing values - if a callback was explicitly set, we don't want to remove it here.